Automattic/jetpack
# Don't sync set_object_terms actions that don't update term relationships

@jmdodd created a fabulous filter to exclude set_object_terms that were not performing updates. 
This patch ensures wp_set_object_terms calls with the same term relations already set on the object will no longer trigger set_object_terms actions to be synced.

Fixes https://wp.me/p9o2xV-12g

#### Changes proposed in this Pull Request:
* set_object_terms actions where the old and new relationships match will not be synced.

#### Jetpack product discussion
N/A

#### Does this pull request change what data or activity we track or use?
NO

#### Testing instructions:
* Call wp_set_object_terms or wp_set_post_terms on an object/post to add relationships
* Verify Sync actions are queued/sent
* repeat the same call from step #1
* Verify no `set_object_terms` action is queued/sent.

** You can see the unit test for examples.

#### Proposed changelog entry for your changes:

* Jetpack Sync : Performance update to minimize set_object_terms actions triggered.
